Manipulator/ServicesManipulator.php

```diff
@@ -99,4 +145,24 @@ public function addResource($serviceId, $modelClass, $adminClass, $controllerNam
             ));
         }
     }
+
+    /**
+     * @param string $servicesContent
+     * @param string $serviceId
+     * @param string $adminClass
+     * @param string $modelClass
+     * @param string $controllerName
+     * @param string $managerType
+     *
+     * @return string
+     */
+    private function createServiceDefinitionXmlNode($servicesContent, $serviceId, $adminClass, $modelClass, $controllerName, $managerType)
+    {
+
+        $template = sprintf($this->xmlServiceDefinitionTemplate, $serviceId, $adminClass, $modelClass, $controllerName, $managerType, current(array_slice(explode('\\', $modelClass), -1)));
+
+        $content = str_replace('</services>', $template, $servicesContent);
```

Comment:
if you load a DOMDocument from a file, append a new node and dump it back to a file, then all line breaks + indentions (4 spaces become 2 spaces) are gone. A services.xml without line breaks between service-definitions is not readable.